addons/binding/org.openhab.binding.rotelra1x/src/main/java/org/openhab/binding/rotelra1x/handler/RotelRa1xHandler.java

```diff
@@ -0,0 +1,345 @@
+                    } catch (InterruptedException e1) {
+                        return;
+                    }
+                }
+            }
+        }
+    }
+
+    private void send(String text) throws IOException, ConfigurationError {
+        if (power) {
+            connect();
+            serialPort.getOutputStream().write(text.getBytes(StandardCharsets.US_ASCII));
+        }
+    }
+
+    private void sendForce(String text) throws IOException, ConfigurationError {
+        connect();
+        serialPort.getOutputStream().write(text.getBytes(StandardCharsets.US_ASCII));
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        try {
+            if (CHANNEL_POWER.equals(channelUID.getId())) {
+                if (command == OnOffType.ON) {
+                    sendForce("power_on!");
+                } else if (command == OnOffType.OFF) {
+                    sendForce("power_off!");
+                } else if (command instanceof RefreshType) {
+                    sendForce("get_current_power!");
+                }
+            } else if (CHANNEL_MUTE.equals(channelUID.getId())) {
+                if (command == OnOffType.ON) {
+                    send("mute_on!");
+                } else {
+                    send("mute_off!");
+                }
+            } else if (CHANNEL_VOLUME.equals(channelUID.getId())) {
+                handleVolume(command);
+            } else if (CHANNEL_BRIGHTNESS.equals(channelUID.getId())) {
+                handleBrightness(command);
+            } else if (CHANNEL_SOURCE.equals(channelUID.getId())) {
+                if (command instanceof StringType) {
+                    send(command.toString() + "!");
+                } else {
+                    send("get_current_source!");
+                }
+            }
+        } catch (IOException e) {
+            logger.debug("An I/O error occurred while processing the command {}.", command, e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            disconnect();
+        } catch (ConfigurationError e) {
+            logger.debug("There is an error in the configuration of the thing.", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            disconnect();
+        }
+    }
+
+    private void handleVolume(Command command) throws IOException, ConfigurationError {
+        if (command instanceof IncreaseDecreaseType && command == IncreaseDecreaseType.INCREASE) {
+            send("volume_up!");
+        } else if (command instanceof IncreaseDecreaseType && command == IncreaseDecreaseType.DECREASE) {
+            send("volume_down!");
+        } else if (command instanceof DecimalType) {
+            double value = Double.parseDouble(command.toString()) * maximumVolume / 100.0;
+            send("volume_" + Integer.toString((int) value) + "!");
```

Comment:
No need to do the `Integer.toString`. This is automatically done. You can just use: `((int) value)`